In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
%matplotlib inline

In [3]:
restaurants_df = pd.read_csv("../data/DOHMH_New_York_City_Restaurant_Inspection_Results.csv")

In [4]:
restaurants_df.head()

CAMIS Restaurant_Name    Borough BUILDING           STREET  \
0  41510938         AFTER 8     Queens     2587  STEINWAY STREET   
1  41692932      GOURMET 45  Manhattan      721         3 AVENUE   
2  50095624     AVENUE CAFE     Queens     3527         30TH AVE   
3  50054381         JUPIOCA   Brooklyn     278B      WYCKOFF AVE   
4  50054381         JUPIOCA   Brooklyn     278B      WYCKOFF AVE   

         Street_Address      Zip       Phone             Cuisine_Description  \
0  2587 STEINWAY STREET  11103.0  7187775757                        American   
1          721 3 AVENUE  10017.0  2126826220                      Sandwiches   
2         3527 30TH AVE  11103.0  7184339298                   Mediterranean   
3      278B WYCKOFF AVE  11237.0  9179826111  Juice, Smoothies, Fruit Salads   
4      278B WYCKOFF AVE  11237.0  9179826111  Juice, Smoothies, Fruit Salads   

  Inspection_Date  ... Record_Date                           Inspection_Type  \
0       3/25/2022  ...   4/30/2022   Smoke-Free Air Act / Initial Inspection   
1       9/26/2019  ...   4/30/2022          Cycle Inspection / Re-inspection   
2       8/28/2019  ...   4/30/2022  Pre-permit (Operational) / Re-inspection   
3        2/5/2019  ...   4/30/2022     Cycle Inspection / Initial Inspection   
4        2/5/2019  ...   4/30/2022     Cycle Inspection / Initial Inspection   

    Latitude  Longitude  Community_District Council_District Census_Tract  \
0  40.767510 -73.912032               401.0             22.0      14300.0   
1  40.752776 -73.972966               106.0              4.0       9000.0   
2  40.765269 -73.917916               401.0             22.0       6502.0   
3  40.700019 -73.911988               304.0             37.0      43900.0   
4  40.700019 -73.911988               304.0             37.0      43900.0   

         BIN           BBL   NTA  
0  4000000.0  4.006840e+09  QN70  
1  1037582.0  1.013190e+09  MN19  
2  4009223.0  4.006290e+09  QN70  
3        NaN  3.000000e+00  BK77  
4        NaN  3.000000e+00  BK77  

[5 rows x 27 columns]

In [5]:
restaurants_df.info

<bound method DataFrame.info of            CAMIS             Restaurant_Name    Borough BUILDING  \
0       41510938                     AFTER 8     Queens     2587   
1       41692932                  GOURMET 45  Manhattan      721   
2       50095624                 AVENUE CAFE     Queens     3527   
3       50054381                     JUPIOCA   Brooklyn     278B   
4       50054381                     JUPIOCA   Brooklyn     278B   
...          ...                         ...        ...      ...   
301725  41710782  FORNO PIZZERIA & TRATTORIA     Queens    52-27   
301726  40837605         BRICK OVEN PIZZA 33  Manhattan      201   
301727  40699061                   I.O. CAFE     Queens      793   
301728  41474882                        BUKA   Brooklyn      946   
301729  41598329           BOSTON ROAD CHINA      Bronx     3483   

                  STREET        Street_Address      Zip       Phone  \
0        STEINWAY STREET  2587 STEINWAY STREET  11103.0  7187775757   
1        

In [6]:
restaurants_df.shape

(301730, 27)

In [7]:
print(restaurants_df.dtypes)

CAMIS                      int64
Restaurant_Name           object
Borough                   object
BUILDING                  object
STREET                    object
Street_Address            object
Zip                      float64
Phone                     object
Cuisine_Description       object
Inspection_Date           object
Action                    object
Violation_Code            object
Violation_Description     object
Critical_Flag             object
Score                    float64
Grade                     object
Grade_Date                object
Record_Date               object
Inspection_Type           object
Latitude                 float64
Longitude                float64
Community_District       float64
Council_District         float64
Census_Tract             float64
BIN                      float64
BBL                      float64
NTA                       object
dtype: object


In [8]:
restaurants_df.describe()

CAMIS            Zip          Score       Latitude  \
count  3.017300e+05  297420.000000  287604.000000  301424.000000   
mean   4.698869e+07   10685.633619      20.906142      40.149078   
std    4.214679e+06     594.593286      15.157427       4.832698   
min    3.007544e+07    7004.000000       0.000000       0.000000   
25%    4.157331e+07   10023.000000      11.000000      40.687063   
50%    5.003744e+07   10470.000000      16.000000      40.734697   
75%    5.007232e+07   11230.000000      27.000000      40.762856   
max    5.012212e+07   20147.000000     157.000000      40.912822   

           Longitude  Community_District  Council_District   Census_Tract  \
count  301424.000000       296735.000000     296742.000000  296742.000000   
mean      -72.885122          250.011933         20.026346   29200.804942   
std         8.772538          130.244462         15.591325   30665.756178   
min       -74.249101          101.000000          1.000000     100.000000   
25%       -73.988473          106.000000          4.000000    8100.000000   
50%       -73.955679          301.000000         20.000000   17000.000000   
75%       -73.895481          401.000000         34.000000   40500.000000   
max         0.000000          595.000000         51.000000  162100.000000   

                BIN           BBL  
count  2.954030e+05  3.010380e+05  
mean   2.525204e+06  2.416485e+09  
std    1.346040e+06  1.338580e+09  
min    1.000000e+06  1.000000e+00  
25%    1.047959e+06  1.010718e+09  
50%    3.009147e+06  3.002170e+09  
75%    4.004099e+06  4.003110e+09  
max    5.799501e+06  5.270001e+09

In [9]:
restaurants_df.corr()

CAMIS       Zip     Score  Latitude  Longitude  \
CAMIS               1.000000  0.048895  0.069668  0.055405  -0.055261   
Zip                 0.048895  1.000000  0.020241 -0.283415   0.249025   
Score               0.069668  0.020241  1.000000  0.017107  -0.017116   
Latitude            0.055405 -0.283415  0.017107  1.000000  -0.999822   
Longitude          -0.055261  0.249025 -0.017116 -0.999822   1.000000   
Community_District  0.024416  0.860966  0.004866 -0.302575   0.171641   
Council_District    0.027312  0.781396  0.012510 -0.206314  -0.002268   
Census_Tract        0.030360  0.571449  0.010534 -0.027666   0.112757   
BIN                 0.026201  0.863928  0.004668 -0.058636   0.024318   
BBL                 0.033226  0.848132  0.009732  0.209984  -0.213227   

                    Community_District  Council_District  Census_Tract  \
CAMIS                         0.024416          0.027312      0.030360   
Zip                           0.860966          0.781396      0.571449   
Score                         0.004866          0.012510      0.010534   
Latitude                     -0.302575         -0.206314     -0.027666   
Longitude                     0.171641         -0.002268      0.112757   
Community_District            1.000000          0.804690      0.502697   
Council_District              0.804690          1.000000      0.379151   
Census_Tract                  0.502697          0.379151      1.000000   
BIN                           0.996497          0.801825      0.506352   
BBL                           0.986733          0.792319      0.490167   

                         BIN       BBL  
CAMIS               0.026201  0.033226  
Zip                 0.863928  0.848132  
Score               0.004668  0.009732  
Latitude           -0.058636  0.209984  
Longitude           0.024318 -0.213227  
Community_District  0.996497  0.986733  
Council_District    0.801825  0.792319  
Census_Tract        0.506352  0.490167  
BIN                 1.000000  0.997420  
BBL                 0.997420  1.000000

Let's the raw value count of restaurants in each bourough (Without filtering for NULLs)

In [10]:
restaurants_df.Borough.value_counts()

Manhattan        116240
Brooklyn          75037
Queens            71228
Bronx             29501
Staten Island      9647
0                    77
Name: Borough, dtype: int64

In [11]:
boro_restaurants = pd.DataFrame(restaurants_df.Borough.value_counts().reset_index())

In [12]:
boro_restaurants

index  Borough
0      Manhattan   116240
1       Brooklyn    75037
2         Queens    71228
3          Bronx    29501
4  Staten Island     9647
5              0       77

In [13]:
boro_restaurants.columns = ['Borough', 'Restaurants']

In [14]:
boro_restaurants

Borough  Restaurants
0      Manhattan       116240
1       Brooklyn        75037
2         Queens        71228
3          Bronx        29501
4  Staten Island         9647
5              0           77

Based of raw data, Manhattan has the most restaurants. There are 77 restaurants where the borough is unknown/NULL

Let's look at the number of restaurants by community district

In [15]:
restaurants_df.Community_District.value_counts()

105.0    22956
103.0    14985
102.0    13976
104.0    12223
407.0    10859
         ...  
164.0       30
227.0       17
226.0       15
595.0       11
482.0        4
Name: Community_District, Length: 69, dtype: int64

In [16]:
CD_restaurants = pd.DataFrame(restaurants_df.Community_District.value_counts().reset_index())

In [17]:
CD_restaurants

index  Community_District
0   105.0               22956
1   103.0               14985
2   102.0               13976
3   104.0               12223
4   407.0               10859
..    ...                 ...
64  164.0                  30
65  227.0                  17
66  226.0                  15
67  595.0                  11
68  482.0                   4

[69 rows x 2 columns]

In [18]:
CD_restaurants.columns = ['Community District', 'Restaurants']

In [19]:
CD_restaurants

Community District  Restaurants
0                105.0        22956
1                103.0        14985
2                102.0        13976
3                104.0        12223
4                407.0        10859
..                 ...          ...
64               164.0           30
65               227.0           17
66               226.0           15
67               595.0           11
68               482.0            4

[69 rows x 2 columns]

Manhattan Community District 105 has the most restaurants.

Let's look at types of restaurants in the city.

In [20]:
restaurants_df.Cuisine_Description.value_counts()

American          55759
Chinese           31483
Pizza             18120
Coffee/Tea        14525
Latin American    12984
                  ...  
Basque               10
Lebanese             10
New French            3
Armenian              3
Bar/Club              1
Name: Cuisine_Description, Length: 88, dtype: int64

In [21]:
cuisine_nyc = pd.DataFrame(restaurants_df.Cuisine_Description.value_counts().reset_index())

In [22]:
cuisine_nyc

index  Cuisine_Description
0         American                55759
1          Chinese                31483
2            Pizza                18120
3       Coffee/Tea                14525
4   Latin American                12984
..             ...                  ...
83          Basque                   10
84        Lebanese                   10
85      New French                    3
86        Armenian                    3
87        Bar/Club                    1

[88 rows x 2 columns]

In [23]:
cuisine_nyc.columns = ['Cuisine Type', 'Restaurants']

In [24]:
cuisine_nyc

Cuisine Type  Restaurants
0         American        55759
1          Chinese        31483
2            Pizza        18120
3       Coffee/Tea        14525
4   Latin American        12984
..             ...          ...
83          Basque           10
84        Lebanese           10
85      New French            3
86        Armenian            3
87        Bar/Club            1

[88 rows x 2 columns]

American food is the most popular restaurant (numerically) in NYC.

Let's look at the health inspection grades.

In [25]:
restaurants_df.Grade.value_counts()

A    116392
B     17385
C      7268
N      5318
Z      2821
P      1700
G         1
Name: Grade, dtype: int64

In [26]:
restaurant_grade_nyc = pd.DataFrame(restaurants_df.Grade.value_counts().reset_index())

In [27]:
restaurant_grade_nyc

index   Grade
0     A  116392
1     B   17385
2     C    7268
3     N    5318
4     Z    2821
5     P    1700
6     G       1

In [28]:
restaurant_grade_nyc.columns = ['Grade', 'Total']

In [29]:
restaurant_grade_nyc

Grade   Total
0     A  116392
1     B   17385
2     C    7268
3     N    5318
4     Z    2821
5     P    1700
6     G       1

Most Restaurants in the city got an A grade. That's good to know.

How much Null data do we have?

In [30]:
restaurants_df.isnull().sum()

CAMIS                         0
Restaurant_Name            1381
Borough                       0
BUILDING                    477
STREET                        7
Street_Address                0
Zip                        4310
Phone                        28
Cuisine_Description        2696
Inspection_Date               0
Action                     4048
Violation_Code             7248
Violation_Description      5517
Critical_Flag                 0
Score                     14126
Grade                    150845
Grade_Date               156150
Record_Date                   0
Inspection_Type            4048
Latitude                    306
Longitude                   306
Community_District         4995
Council_District           4988
Census_Tract               4988
BIN                        6327
BBL                         692
NTA                        4996
dtype: int64

There's a decent amount of NULL values in this database. Some of it is located in columns we will not need for our specific analysis.

Let's make some dataframes based on boroughs and the Health Food Analysis notebook.

Let's start with the boroughs first.

In [31]:
brooklyn_df = restaurants_df[restaurants_df.Borough == 'Brooklyn']

In [32]:
brooklyn_df.head()

CAMIS          Restaurant_Name   Borough BUILDING           STREET  \
3  50054381                  JUPIOCA  Brooklyn     278B      WYCKOFF AVE   
4  50054381                  JUPIOCA  Brooklyn     278B      WYCKOFF AVE   
5  50088685        FLAVOR RESTAURANT  Brooklyn     7109        13 AVENUE   
6  50088685        FLAVOR RESTAURANT  Brooklyn     7109        13 AVENUE   
7  40604442  LA LECHONERA RESTAURANT  Brooklyn      747  NEW LOTS AVENUE   

        Street_Address      Zip       Phone             Cuisine_Description  \
3     278B WYCKOFF AVE  11237.0  9179826111  Juice, Smoothies, Fruit Salads   
4     278B WYCKOFF AVE  11237.0  9179826111  Juice, Smoothies, Fruit Salads   
5       7109 13 AVENUE  11228.0  7182324680                         Chinese   
6       7109 13 AVENUE  11228.0  7182324680                         Chinese   
7  747 NEW LOTS AVENUE  11207.0  3476361224                  Latin American   

  Inspection_Date  ... Record_Date  \
3        2/5/2019  ...   4/30/2022   
4        2/5/2019  ...   4/30/2022   
5       2/22/2019  ...   4/30/2022   
6       2/22/2019  ...   4/30/2022   
7       4/29/2019  ...   4/30/2022   

                                   Inspection_Type   Latitude  Longitude  \
3            Cycle Inspection / Initial Inspection  40.700019 -73.911988   
4            Cycle Inspection / Initial Inspection  40.700019 -73.911988   
5  Pre-permit (Operational) / Reopening Inspection  40.622455 -74.005169   
6  Pre-permit (Operational) / Reopening Inspection  40.622455 -74.005169   
7     Administrative Miscellaneous / Re-inspection  40.666021 -73.883528   

   Community_District Council_District Census_Tract        BIN           BBL  \
3               304.0             37.0      43900.0        NaN  3.000000e+00   
4               304.0             37.0      43900.0        NaN  3.000000e+00   
5               310.0             43.0      19600.0  3157655.0  3.061780e+09   
6               310.0             43.0      19600.0  3157655.0  3.061780e+09   
7               305.0             42.0     112400.0  3090737.0  3.040930e+09   

    NTA  
3  BK77  
4  BK77  
5  BK30  
6  BK30  
7  BK82  

[5 rows x 27 columns]

In [33]:
brooklyn = pd.DataFrame(brooklyn_df)

In [34]:
brooklyn

CAMIS             Restaurant_Name   Borough BUILDING  \
3       50054381                     JUPIOCA  Brooklyn     278B   
4       50054381                     JUPIOCA  Brooklyn     278B   
5       50088685           FLAVOR RESTAURANT  Brooklyn     7109   
6       50088685           FLAVOR RESTAURANT  Brooklyn     7109   
7       40604442     LA LECHONERA RESTAURANT  Brooklyn      747   
...          ...                         ...       ...      ...   
301710  50065187              TACOS EL NOPAL  Brooklyn     8321   
301720  50045504  TABLE 40 PIZZERIA AND CAFE  Brooklyn     4001   
301721  50088657                MINNIE'S BAR  Brooklyn      885   
301724  41480427     YEMEN CAFE & RESTAURANT  Brooklyn      176   
301728  41474882                        BUKA  Brooklyn      946   

                 STREET       Street_Address      Zip       Phone  \
3           WYCKOFF AVE     278B WYCKOFF AVE  11237.0  9179826111   
4           WYCKOFF AVE     278B WYCKOFF AVE  11237.0  9179826111   
5             13 AVENUE       7109 13 AVENUE  11228.0  7182324680   
6             13 AVENUE       7109 13 AVENUE  11228.0  7182324680   
7       NEW LOTS AVENUE  747 NEW LOTS AVENUE  11207.0  3476361224   
...                 ...                  ...      ...         ...   
301710         5 AVENUE        8321 5 AVENUE  11209.0  3475787461   
301720        13 AVENUE       4001 13 AVENUE  11218.0  9292345444   
301721         4 AVENUE         885 4 AVENUE  11232.0  2405350815   
301724  ATLANTIC AVENUE  176 ATLANTIC AVENUE  11201.0  7188349533   
301728    FULTON STREET    946 FULTON STREET  11238.0  3477630619   

                   Cuisine_Description Inspection_Date  ... Record_Date  \
3       Juice, Smoothies, Fruit Salads        2/5/2019  ...   4/30/2022   
4       Juice, Smoothies, Fruit Salads        2/5/2019  ...   4/30/2022   
5                              Chinese       2/22/2019  ...   4/30/2022   
6                              Chinese       2/22/2019  ...   4/30/2022   
7                       Latin American       4/29/2019  ...   4/30/2022   
...                                ...             ...  ...         ...   
301710                         Tex-Mex       5/20/2019  ...   4/30/2022   
301720                   Jewish/Kosher      11/16/2016  ...   4/30/2022   
301721                           Other        4/2/2019  ...   4/30/2022   
301724                  Middle Eastern       6/11/2019  ...   4/30/2022   
301728                         African       12/6/2018  ...   4/30/2022   

                                        Inspection_Type   Latitude  Longitude  \
3                 Cycle Inspection / Initial Inspection  40.700019 -73.911988   
4                 Cycle Inspection / Initial Inspection  40.700019 -73.911988   
5       Pre-permit (Operational) / Reopening Inspection  40.622455 -74.005169   
6       Pre-permit (Operational) / Reopening Inspection  40.622455 -74.005169   
7          Administrative Miscellaneous / Re-inspection  40.666021 -73.883528   
...                                                 ...        ...        ...   
301710                 Cycle Inspection / Re-inspection  40.623366 -74.025122   
301720                 Cycle Inspection / Re-inspection  40.639971 -73.986956   
301721    Pre-permit (Operational) / Initial Inspection  40.655845 -74.002808   
301724                 Cycle Inspection / Re-inspection  40.689998 -73.993538   
301728                 Cycle Inspection / Re-inspection  40.682680 -73.963130   

        Community_District Council_District Census_Tract        BIN  \
3                    304.0             37.0      43900.0        NaN   
4                    304.0             37.0      43900.0        NaN   
5                    310.0             43.0      19600.0  3157655.0   
6                    310.0             43.0      19600.0  3157655.0   
7                    305.0             42.0     112400.0  3090737.0   
...                    ...              ...          ...        ...   
301710

In [35]:
brooklyn_df.Cuisine_Description.value_counts()

American        12207
Chinese          8958
Caribbean        5117
Pizza            4558
Coffee/Tea       3944
                ...  
Indonesian         12
New American       11
Scandinavian        5
English             5
Lebanese            1
Name: Cuisine_Description, Length: 78, dtype: int64

In [36]:
brooklyn_cuisine = pd.DataFrame(brooklyn_df.Cuisine_Description.value_counts().reset_index())

In [37]:
brooklyn_cuisine

index  Cuisine_Description
0       American                12207
1        Chinese                 8958
2      Caribbean                 5117
3          Pizza                 4558
4     Coffee/Tea                 3944
..           ...                  ...
73    Indonesian                   12
74  New American                   11
75  Scandinavian                    5
76       English                    5
77      Lebanese                    1

[78 rows x 2 columns]

In [38]:
brooklyn_cuisine.columns = ['Cuisine Type', 'Restaurants']

In [39]:
brooklyn_cuisine

Cuisine Type  Restaurants
0       American        12207
1        Chinese         8958
2      Caribbean         5117
3          Pizza         4558
4     Coffee/Tea         3944
..           ...          ...
73    Indonesian           12
74  New American           11
75  Scandinavian            5
76       English            5
77      Lebanese            1

[78 rows x 2 columns]

American restaurants make up most of the restaurants in this borough.

In [40]:
brooklyn_df.Grade.value_counts()

A    28381
B     4238
C     1741
N     1494
Z      811
P      481
G        1
Name: Grade, dtype: int64

In [41]:
brooklyn_grades = pd.DataFrame(brooklyn_df.Grade.value_counts().reset_index())

In [42]:
brooklyn_grades

index  Grade
0     A  28381
1     B   4238
2     C   1741
3     N   1494
4     Z    811
5     P    481
6     G      1

In [43]:
brooklyn_grades.columns = ['Grade', 'Total']

In [44]:
brooklyn_grades

Grade  Total
0     A  28381
1     B   4238
2     C   1741
3     N   1494
4     Z    811
5     P    481
6     G      1

The majority of restaurants in this borough recieved a grade of A from the health inspector.

In [45]:
bronx_df = restaurants_df[restaurants_df.Borough == 'Bronx']

In [46]:
bronx_df.head()

CAMIS                          Restaurant_Name Borough BUILDING  \
16  50067278  MOFONGO EL MOCANO RESTAURANT AND LOUNGE   Bronx      985   
43  50034610                   MAMA G AFRICAN KITCHEN   Bronx    3650A   
58  41195691                    F & J PINE RESTAURANT   Bronx     1913   
61  50070647                   SAN JOSE TULCINGO DELI   Bronx      109   
69  50066844                               MCDONALD'S   Bronx     1101   

                 STREET            Street_Address      Zip       Phone  \
16        MORRIS AVENUE         985 MORRIS AVENUE  10456.0  7185885103   
43    WHITE PLAINS ROAD   3650A WHITE PLAINS ROAD  10467.0  9292224813   
58     BRONXDALE AVENUE     1913 BRONXDALE AVENUE  10462.0  7187925956   
61     EAST  153 STREET      109 EAST  153 STREET  10451.0  9176319253   
69  EAST TREMONT AVENUE  1101 EAST TREMONT AVENUE  10460.0  7188244123   

   Cuisine_Description Inspection_Date  ... Record_Date  \
16      Latin American       12/5/2018  ...   4/30/2022   
43             African      10/16/2018  ...   4/30/2022   
58             Italian       9/25/2017  ...   4/30/2022   
61             Mexican       11/6/2017  ...   4/30/2022   
69          Hamburgers       7/31/2017  ...   4/30/2022   

                                  Inspection_Type   Latitude  Longitude  \
16          Cycle Inspection / Initial Inspection  40.828499 -73.916854   
43               Cycle Inspection / Re-inspection  40.880317 -73.864654   
58          Cycle Inspection / Initial Inspection  40.848944 -73.862061   
61  Pre-permit (Operational) / Initial Inspection  40.823926 -73.928032   
69  Pre-permit (Operational) / Initial Inspection  40.839991 -73.877118   

    Community_District Council_District Census_Tract        BIN           BBL  \
16               204.0             16.0      18302.0  2002524.0  2.024470e+09   
43               212.0             12.0      38000.0  2095077.0  2.046620e+09   
58               211.0             13.0      24600.0  2048074.0  2.042630e+09   
61               204.0              8.0       6300.0  2002968.0  2.024820e+09   
69               206.0             15.0      22000.0  2042429.0  2.040040e+09   

     NTA  
16  BX14  
43  BX44  
58  BX49  
61  BX63  
69  BX17  

[5 rows x 27 columns]

In [47]:
bronx = pd.DataFrame(bronx_df)

In [48]:
bronx

CAMIS                          Restaurant_Name Borough BUILDING  \
16      50067278  MOFONGO EL MOCANO RESTAURANT AND LOUNGE   Bronx      985   
43      50034610                   MAMA G AFRICAN KITCHEN   Bronx    3650A   
58      41195691                    F & J PINE RESTAURANT   Bronx     1913   
61      50070647                   SAN JOSE TULCINGO DELI   Bronx      109   
69      50066844                               MCDONALD'S   Bronx     1101   
...          ...                                      ...     ...      ...   
301690  41164361                       BEST ITALIAN PIZZA   Bronx  200-202   
301693  50104499                 SHAWATI ADEN RESTAURANTS   Bronx      675   
301696  41545589               ADDEO'S RIVERDALE PIZZERIA   Bronx     5654   
301712  50086535                                 JUICEADE   Bronx     1345   
301729  41598329                        BOSTON ROAD CHINA   Bronx     3483   

                     STREET             Street_Address      Zip       Phone  \
16            MORRIS AVENUE          985 MORRIS AVENUE  10456.0  7185885103   
43        WHITE PLAINS ROAD    3650A WHITE PLAINS ROAD  10467.0  9292224813   
58         BRONXDALE AVENUE      1913 BRONXDALE AVENUE  10462.0  7187925956   
61         EAST  153 STREET       109 EAST  153 STREET  10451.0  9176319253   
69      EAST TREMONT AVENUE   1101 EAST TREMONT AVENUE  10460.0  7188244123   
...                     ...                        ...      ...         ...   
301690    EAST FORDHAM ROAD  200-202 EAST FORDHAM ROAD  10458.0  7185625806   
301693   MORRIS PARK AVENUE     675 MORRIS PARK AVENUE  10462.0  6463823660   
301696     RIVERDALE AVENUE      5654 RIVERDALE AVENUE  10471.0  7185494750   
301712   SOUTHERN BOULEVARD    1345 SOUTHERN BOULEVARD  10459.0  7189916369   
301729          BOSTON ROAD           3483 BOSTON ROAD  10469.0  7188819888   

                   Cuisine_Description Inspection_Date  ... Record_Date  \
16                      Latin American       12/5/2018  ...   4/30/2022   
43                             African      10/16/2018  ...   4/30/2022   
58                             Italian       9/25/2017  ...   4/30/2022   
61                             Mexican       11/6/2017  ...   4/30/2022   
69                          Hamburgers       7/31/2017  ...   4/30/2022   
...                                ...             ...  ...         ...   
301690                           Pizza        9/6/2017  ...   4/30/2022   
301693                  Middle Eastern      10/28/2021  ...   4/30/2022   
301696                           Pizza        2/1/2017  ...   4/30/2022   
301712  Juice, Smoothies, Fruit Salads        4/6/2019  ...   4/30/2022   
301729                         Chinese      12/27/2016  ...   4/30/2022   

                                      Inspection_Type   Latitude  Longitude  \
16              Cycle Inspection / Initial Inspection  40.828499 -73.916854   
43                   Cycle Inspection / Re-inspection  40.880317 -73.864654   
58              Cycle Inspection / Initial Inspection  40.848944 -73.862061   
61      Pre-permit (Operational) / Initial Inspection  40.823926 -73.928032   
69      Pre-permit (Operational) / Initial Inspection  40.839991 -73.877118   
...                                               ...        ...        ...   
301690          Cycle Inspection / Initial Inspection  40.862402 -73.896783   
301693          Cycle Inspection / Initial Inspection  40.845091 -73.866764   
301696          Cycle Inspection / Initial Inspection  40.905559 -73.904472   
301712  Pre-permit (Operational) / Initial Inspection  40.831236 -73.891400   
301729          Cycle Inspection / Initial Inspection  40.876803 -73.847090   

        Community_District Council_District Census_Tract        BIN  \
16                   204.0             16.0      18302.0  2002524.0   
43                   212.0             12.0      38000.0  2095077.0   
58                   211.0             13.0      24600.0  2048

In [49]:
bronx_df.Cuisine_Description.value_counts()

Chinese                           3565
American                          3537
Latin American                    3146
Pizza                             2578
Spanish                           2130
Chicken                           1910
Caribbean                         1898
Mexican                           1816
Hamburgers                         924
Bakery Products/Desserts           825
Italian                            701
Sandwiches                         672
Donuts                             643
African                            583
Coffee/Tea                         515
Juice, Smoothies, Fruit Salads     500
Seafood                            484
Tex-Mex                            420
Japanese                           213
Jewish/Kosher                      163
Irish                              150
Not Listed/Not Applicable          146
Bangladeshi                        138
Other                              132
Frozen Desserts                    131
Middle Eastern           

In [50]:
bronx_cuisine = pd.DataFrame(bronx_df.Cuisine_Description.value_counts().reset_index())

In [51]:
bronx_cuisine

index  Cuisine_Description
0                          Chinese                 3565
1                         American                 3537
2                   Latin American                 3146
3                            Pizza                 2578
4                          Spanish                 2130
5                          Chicken                 1910
6                        Caribbean                 1898
7                          Mexican                 1816
8                       Hamburgers                  924
9         Bakery Products/Desserts                  825
10                         Italian                  701
11                      Sandwiches                  672
12                          Donuts                  643
13                         African                  583
14                      Coffee/Tea                  515
15  Juice, Smoothies, Fruit Salads                  500
16                         Seafood                  484
17                         Tex-Mex                  420
18                        Japanese                  213
19                   Jewish/Kosher                  163
20                           Irish                  150
21       Not Listed/Not Applicable                  146
22                     Bangladeshi                  138
23                           Other                  132
24                 Frozen Desserts                  131
25                  Middle Eastern                  131
26              Asian/Asian Fusion                  128
27                 Bagels/Pretzels                  112
28                   Mediterranean                  110
29               Bottled Beverages                   82
30                          Indian                   81
31                            Thai                   79
32  Sandwiches/Salads/Mixed Buffet                   77
33                           Greek                   71
34                       Soul Food                   71
35                Pancakes/Waffles                   66
36                      Steakhouse                   61
37                 Southeast Asian                   55
38         Soups/Salads/Sandwiches                   48
39                        Barbecue                   40
40                        Peruvian                   40
41                Hotdogs/Pretzels                   33
42                   Chinese/Cuban                   26
43                Eastern European                   20
44                         Hotdogs                   12
45                          Fusion                    8
46                           Czech                    6
47                          French                    6
48                      Vegetarian                    5
49                           Cajun                    5
50                       Pakistani                    3

In [52]:
bronx_cuisine.columns = ['Cuisine Type', 'Restaurants']

In [53]:
bronx_cuisine

Cuisine Type  Restaurants
0                          Chinese         3565
1                         American         3537
2                   Latin American         3146
3                            Pizza         2578
4                          Spanish         2130
5                          Chicken         1910
6                        Caribbean         1898
7                          Mexican         1816
8                       Hamburgers          924
9         Bakery Products/Desserts          825
10                         Italian          701
11                      Sandwiches          672
12                          Donuts          643
13                         African          583
14                      Coffee/Tea          515
15  Juice, Smoothies, Fruit Salads          500
16                         Seafood          484
17                         Tex-Mex          420
18                        Japanese          213
19                   Jewish/Kosher          163
20                           Irish          150
21       Not Listed/Not Applicable          146
22                     Bangladeshi          138
23                           Other          132
24                 Frozen Desserts          131
25                  Middle Eastern          131
26              Asian/Asian Fusion          128
27                 Bagels/Pretzels          112
28                   Mediterranean          110
29               Bottled Beverages           82
30                          Indian           81
31                            Thai           79
32  Sandwiches/Salads/Mixed Buffet           77
33                           Greek           71
34                       Soul Food           71
35                Pancakes/Waffles           66
36                      Steakhouse           61
37                 Southeast Asian           55
38         Soups/Salads/Sandwiches           48
39                        Barbecue           40
40                        Peruvian           40
41                Hotdogs/Pretzels           33
42                   Chinese/Cuban           26
43                Eastern European           20
44                         Hotdogs           12
45                          Fusion            8
46                           Czech            6
47                          French            6
48                      Vegetarian            5
49                           Cajun            5
50                       Pakistani            3

Chinese restaurants make up most of the restaurants in this borough.

In [54]:
bronx_df.Grade.value_counts()

A    10972
B     2158
C      800
N      514
Z      254
P      185
Name: Grade, dtype: int64

In [55]:
bronx_grades = pd.DataFrame(bronx_df.Grade.value_counts().reset_index())

In [56]:
bronx_grades

index  Grade
0     A  10972
1     B   2158
2     C    800
3     N    514
4     Z    254
5     P    185

In [57]:
bronx_grades.columns = ['Grade', 'Total']

In [58]:
bronx_grades

Grade  Total
0     A  10972
1     B   2158
2     C    800
3     N    514
4     Z    254
5     P    185

The majority of restaurants in this borough recieved a grade of A from the health inspector.

In [59]:
manhattan_df = restaurants_df[restaurants_df.Borough == 'Manhattan']

In [60]:
manhattan_df.head()

CAMIS         Restaurant_Name    Borough BUILDING            STREET  \
1   41692932              GOURMET 45  Manhattan      721          3 AVENUE   
8   50034379              BLACK BARN  Manhattan       19  EAST   26 STREET   
9   50063233  FRESNO II GOURMET DELI  Manhattan       31          AVENUE C   
11  40994221                  MANGIA  Manhattan       22  WEST   23 STREET   
12  41663204                54 BELOW  Manhattan      254  WEST   54 STREET   

          Street_Address      Zip       Phone Cuisine_Description  \
1           721 3 AVENUE  10017.0  2126826220          Sandwiches   
8    19 EAST   26 STREET  10010.0  2122655959            American   
9            31 AVENUE C  10009.0  2122531500          Sandwiches   
11   22 WEST   23 STREET  10010.0  2126470287             Italian   
12  254 WEST   54 STREET  10019.0  6464763551            American   

   Inspection_Date  ... Record_Date                        Inspection_Type  \
1        9/26/2019  ...   4/30/2022       Cycle Inspection / Re-inspection   
8         2/4/2020  ...   4/30/2022  Cycle Inspection / Initial Inspection   
9         4/5/2019  ...   4/30/2022  Cycle Inspection / Initial Inspection   
11        2/7/2020  ...   4/30/2022  Cycle Inspection / Initial Inspection   
12        1/7/2020  ...   4/30/2022       Cycle Inspection / Re-inspection   

     Latitude  Longitude  Community_District Council_District Census_Tract  \
1   40.752776 -73.972966               106.0              4.0       9000.0   
8   40.743043 -73.987228               105.0              2.0       5600.0   
9   40.721630 -73.980649               103.0              2.0       2202.0   
11  40.741937 -73.990534               105.0              3.0       5800.0   
12  40.764331 -73.983217               105.0              3.0      13100.0   

          BIN           BBL   NTA  
1   1037582.0  1.013190e+09  MN19  
8   1016877.0  1.008568e+09  MN13  
9   1004598.0  1.003850e+09  MN28  
11  1015572.0  1.008240e+09  MN13  
12  1024843.0  1.010250e+09  MN17  

[5 rows x 27 columns]

In [61]:
manhattan = pd.DataFrame(manhattan_df)

In [62]:
manhattan

CAMIS              Restaurant_Name    Borough BUILDING  \
1       41692932                   GOURMET 45  Manhattan      721   
8       50034379                   BLACK BARN  Manhattan       19   
9       50063233       FRESNO II GOURMET DELI  Manhattan       31   
11      40994221                       MANGIA  Manhattan       22   
12      41663204                     54 BELOW  Manhattan      254   
...          ...                          ...        ...      ...   
301718  50038055  GREAT WALL SHI'S RESTAURANT  Manhattan    527-9   
301719  50117281                   JUST SALAD  Manhattan     1350   
301722  50082098          BROOKLYN CHOP HOUSE  Manhattan      150   
301723  50086536                  JUICE PRESS  Manhattan       10   
301726  40837605          BRICK OVEN PIZZA 33  Manhattan      201   

                        STREET               Street_Address      Zip  \
1                     3 AVENUE                 721 3 AVENUE  10017.0   
8             EAST   26 STREET          19 EAST   26 STREET  10010.0   
9                     AVENUE C                  31 AVENUE C  10009.0   
11            WEST   23 STREET          22 WEST   23 STREET  10010.0   
12            WEST   54 STREET         254 WEST   54 STREET  10019.0   
...                        ...                          ...      ...   
301718               LENOX AVE              527-9 LENOX AVE  10037.0   
301719  AVENUE OF THE AMERICAS  1350 AVENUE OF THE AMERICAS  10019.0   
301722           NASSAU STREET            150 NASSAU STREET  10038.0   
301723         COLUMBUS CIRCLE           10 COLUMBUS CIRCLE  10019.0   
301726        EAST   33 STREET         201 EAST   33 STREET  10016.0   

             Phone             Cuisine_Description Inspection_Date  ...  \
1       2126826220                      Sandwiches       9/26/2019  ...   
8       2122655959                        American        2/4/2020  ...   
9       2122531500                      Sandwiches        4/5/2019  ...   
11      2126470287                         Italian        2/7/2020  ...   
12      6464763551                        American        1/7/2020  ...   
...            ...                             ...             ...  ...   
301718  2129268301                         Chinese       9/18/2019  ...   
301719  5165082187                          Salads        1/1/1900  ...   
301722  2126191200                        American       7/10/2019  ...   
301723  6469011318  Juice, Smoothies, Fruit Salads        5/3/2019  ...   
301726  2125459191                           Pizza       8/12/2021  ...   

       Record_Date                               Inspection_Type   Latitude  \
1        4/30/2022              Cycle Inspection / Re-inspection  40.752776   
8        4/30/2022         Cycle Inspection / Initial Inspection  40.743043   
9        4/30/2022         Cycle Inspection / Initial Inspection  40.721630   
11       4/30/2022         Cycle Inspection / Initial Inspection  40.741937   
12       4/30/2022              Cycle Inspection / Re-inspection  40.764331   
...            ...                                           ...        ...   
301718   4/30/2022         Cycle Inspection / Initial Inspection  40.815056   
301719   4/30/2022                                           NaN  40.762725   
301722   4/30/2022         Cycle Inspection / Initial Inspection  40.711534   
301723   4/30/2022  Administrative Miscellaneous / Re-inspection  40.767808   
301726   4/30/2022         Cycle Inspection / Initial Inspection  40.744914   

        Longitude  Community_District Council_District Census_Tract  \
1      -73.972966               106.0              4.0       9000.0   
8      -73.987228               105.0              2.0       5600.0   
9      -73.980649               103.0              2.0       2202.0   
11     -73.990534               105.0              3.0       5800.0   
12     -73.983217               105.0              3.0      13100.0   
...           ...                 ...    

In [63]:
manhattan_df.Cuisine_Description.value_counts()

American        28561
Chinese          7736
Coffee/Tea       7508
Italian          7263
Japanese         6232
                ...  
Fusion              9
Creole/Cajun        5
New French          3
Armenian            3
Bar/Club            1
Name: Cuisine_Description, Length: 85, dtype: int64

In [64]:
manhattan_cuisine = pd.DataFrame(manhattan_df.Cuisine_Description.value_counts().reset_index())

In [65]:
manhattan_cuisine

index  Cuisine_Description
0       American                28561
1        Chinese                 7736
2     Coffee/Tea                 7508
3        Italian                 7263
4       Japanese                 6232
..           ...                  ...
80        Fusion                    9
81  Creole/Cajun                    5
82    New French                    3
83      Armenian                    3
84      Bar/Club                    1

[85 rows x 2 columns]

In [66]:
manhattan_cuisine.columns = ['Cuisine Type', 'Restaurants']

In [67]:
manhattan_cuisine

Cuisine Type  Restaurants
0       American        28561
1        Chinese         7736
2     Coffee/Tea         7508
3        Italian         7263
4       Japanese         6232
..           ...          ...
80        Fusion            9
81  Creole/Cajun            5
82    New French            3
83      Armenian            3
84      Bar/Club            1

[85 rows x 2 columns]

American restaurants make up most of the restaurants in this borough.

In [68]:
manhattan_df.Grade.value_counts()

A    45461
B     6338
C     2652
N     1878
Z      982
P      556
Name: Grade, dtype: int64

In [69]:
manhattan_grades = pd.DataFrame(manhattan_df.Grade.value_counts().reset_index())

In [70]:
manhattan_grades

index  Grade
0     A  45461
1     B   6338
2     C   2652
3     N   1878
4     Z    982
5     P    556

In [71]:
manhattan_grades.columns = ['Grade', 'Total']

In [72]:
manhattan_grades

Grade  Total
0     A  45461
1     B   6338
2     C   2652
3     N   1878
4     Z    982
5     P    556

The majority of restaurants in this borough recieved a grade of A from the health inspector.

In [73]:
queens_df = restaurants_df[restaurants_df.Borough == 'Queens']

In [74]:
queens_df.head()

CAMIS           Restaurant_Name Borough BUILDING           STREET  \
0   41510938                   AFTER 8  Queens     2587  STEINWAY STREET   
2   50095624               AVENUE CAFE  Queens     3527         30TH AVE   
18  50009171             AUSTIN PUBLIC  Queens     7028        AUSTIN ST   
19  50000738  D'ANNA'S DELI & PIZZERIA  Queens    16908    CROCHERON AVE   
22  50046201      DOUBLE HAPPY KITCHEN  Queens     8518      JAMAICA AVE   

          Street_Address      Zip       Phone Cuisine_Description  \
0   2587 STEINWAY STREET  11103.0  7187775757            American   
2          3527 30TH AVE  11103.0  7184339298       Mediterranean   
18        7028 AUSTIN ST  11375.0  7185750070               Irish   
19   16908 CROCHERON AVE  11358.0  7188780042      Latin American   
22      8518 JAMAICA AVE  11421.0  7188506003             Chinese   

   Inspection_Date  ... Record_Date                           Inspection_Type  \
0        3/25/2022  ...   4/30/2022   Smoke-Free Air Act / Initial Inspection   
2        8/28/2019  ...   4/30/2022  Pre-permit (Operational) / Re-inspection   
18       2/25/2019  ...   4/30/2022     Cycle Inspection / Initial Inspection   
19        8/8/2019  ...   4/30/2022     Cycle Inspection / Initial Inspection   
22       2/27/2019  ...   4/30/2022     Cycle Inspection / Initial Inspection   

     Latitude  Longitude  Community_District Council_District Census_Tract  \
0   40.767510 -73.912032               401.0             22.0      14300.0   
2   40.765269 -73.917916               401.0             22.0       6502.0   
18  40.720612 -73.846061               406.0             29.0      71100.0   
19  40.762755 -73.796854               407.0             19.0     117500.0   
22  40.692519 -73.858837               409.0             32.0       1000.0   

          BIN           BBL   NTA  
0   4000000.0  4.006840e+09  QN70  
2   4009223.0  4.006290e+09  QN70  
18  4077471.0  4.032340e+09  QN17  
19  4119754.0  4.052960e+09  QN51  
22  4183849.0  4.089220e+09  QN53  

[5 rows x 27 columns]

In [75]:
queens = pd.DataFrame(queens_df)

In [76]:
queens

CAMIS                  Restaurant_Name Borough BUILDING  \
0       41510938                          AFTER 8  Queens     2587   
2       50095624                      AVENUE CAFE  Queens     3527   
18      50009171                    AUSTIN PUBLIC  Queens     7028   
19      50000738         D'ANNA'S DELI & PIZZERIA  Queens    16908   
22      50046201             DOUBLE HAPPY KITCHEN  Queens     8518   
...          ...                              ...     ...      ...   
301704  50057589                   TOUS LES JOURS  Queens     3916   
301714  41031977  BROOKLYN BAGEL & COFFEE COMPANY  Queens     3509   
301716  50046794                     AUNT GINNY'S  Queens      652   
301725  41710782       FORNO PIZZERIA & TRATTORIA  Queens    52-27   
301727  40699061                        I.O. CAFE  Queens      793   

                   STREET          Street_Address      Zip       Phone  \
0         STEINWAY STREET    2587 STEINWAY STREET  11103.0  7187775757   
2                30TH AVE           3527 30TH AVE  11103.0  7184339298   
18              AUSTIN ST          7028 AUSTIN ST  11375.0  7185750070   
19          CROCHERON AVE     16908 CROCHERON AVE  11358.0  7188780042   
22            JAMAICA AVE        8518 JAMAICA AVE  11421.0  7188506003   
...                   ...                     ...      ...         ...   
301704          PRINCE ST          3916 PRINCE ST  11354.0  7188881992   
301714  DITMARS BOULEVARD  3509 DITMARS BOULEVARD  11105.0  7189328280   
301716    WOODWARD AVENUE     652 WOODWARD AVENUE  11385.0  3474898899   
301725          69 STREET         52-27 69 STREET  11378.0  7184241200   
301727    FAIRVIEW AVENUE     793 FAIRVIEW AVENUE  11385.0  7184561207   

             Cuisine_Description Inspection_Date  ... Record_Date  \
0                       American       3/25/2022  ...   4/30/2022   
2                  Mediterranean       8/28/2019  ...   4/30/2022   
18                         Irish       2/25/2019  ...   4/30/2022   
19                Latin American        8/8/2019  ...   4/30/2022   
22                       Chinese       2/27/2019  ...   4/30/2022   
...                          ...             ...  ...         ...   
301704  Bakery Products/Desserts       8/29/2019  ...   4/30/2022   
301714           Bagels/Pretzels       1/18/2022  ...   4/30/2022   
301716                  American      11/21/2019  ...   4/30/2022   
301725                     Pizza       3/21/2019  ...   4/30/2022   
301727                    Polish       3/26/2019  ...   4/30/2022   

                                 Inspection_Type   Latitude  Longitude  \
0        Smoke-Free Air Act / Initial Inspection  40.767510 -73.912032   
2       Pre-permit (Operational) / Re-inspection  40.765269 -73.917916   
18         Cycle Inspection / Initial Inspection  40.720612 -73.846061   
19         Cycle Inspection / Initial Inspection  40.762755 -73.796854   
22         Cycle Inspection / Initial Inspection  40.692519 -73.858837   
...                                          ...        ...        ...   
301704     Cycle Inspection / Initial Inspection  40.759478 -73.832243   
301714     Cycle Inspection / Initial Inspection  40.774756 -73.908748   
301716          Cycle Inspection / Re-inspection  40.705644 -73.906989   
301725     Cycle Inspection / Initial Inspection  40.732728 -73.895888   
301727          Cycle Inspection / Re-inspection  40.704705 -73.902446   

        Community_District Council_District Census_Tract        BIN  \
0                    401.0             22.0      14300.0  4000000.0   
2                    401.0             22.0       6502.0  4009223.0   
18                   406.0             29.0      71100.0  4077471.0   
19                   407.0             19.0     117500.0  4119754.0   
22                   409.0             32.0       1000.0  4183849.0   
...                    ...              ...          ...        ...   
301704               407.0             20.0      87100.0  4539244.

In [77]:
queens_df.Cuisine_Description.value_counts()

Chinese              10276
American              9558
Latin American        4948
Pizza                 4059
Spanish               3083
                     ...  
Fruits/Vegetables        8
English                  7
Scandinavian             6
Chinese/Cuban            5
Cajun                    3
Name: Cuisine_Description, Length: 77, dtype: int64

In [78]:
queens_cuisine = pd.DataFrame(queens_df.Cuisine_Description.value_counts().reset_index())

In [79]:
queens_cuisine

index  Cuisine_Description
0             Chinese                10276
1            American                 9558
2      Latin American                 4948
3               Pizza                 4059
4             Spanish                 3083
..                ...                  ...
72  Fruits/Vegetables                    8
73            English                    7
74       Scandinavian                    6
75      Chinese/Cuban                    5
76              Cajun                    3

[77 rows x 2 columns]

In [80]:
queens_cuisine.columns = ['Cuisine Type', 'Restaurants']

In [81]:
queens_cuisine

Cuisine Type  Restaurants
0             Chinese        10276
1            American         9558
2      Latin American         4948
3               Pizza         4059
4             Spanish         3083
..                ...          ...
72  Fruits/Vegetables            8
73            English            7
74       Scandinavian            6
75      Chinese/Cuban            5
76              Cajun            3

[77 rows x 2 columns]

Chinese restaurants make up most of the restaurants in this borough.

In [82]:
queens_df.Grade.value_counts()

A    27345
B     4161
C     1901
N     1268
Z      711
P      435
Name: Grade, dtype: int64

In [83]:
queens_grades = pd.DataFrame(queens_df.Grade.value_counts().reset_index())

In [84]:
queens_grades

index  Grade
0     A  27345
1     B   4161
2     C   1901
3     N   1268
4     Z    711
5     P    435

In [85]:
queens_grades.columns = ['Grade', 'Total']

In [86]:
queens_grades

Grade  Total
0     A  27345
1     B   4161
2     C   1901
3     N   1268
4     Z    711
5     P    435

The majority of restaurants in this borough recieved a grade of A from the health inspector.

In [87]:
si_df = restaurants_df[restaurants_df.Borough == 'Staten Island']

In [88]:
si_df.head()

CAMIS                     Restaurant_Name        Borough BUILDING  \
40   50106984               KENNEDY FRIED CHICKEN  Staten Island      480   
156  40387243                JIMMY MAX RESTAURANT  Staten Island      280   
162  50094799  BOTANAS MEXICAN ICE CREAM & SNACKS  Staten Island      463   
236  50016070      THE STONE HOUSE AT CLOVE LAKES  Staten Island     1150   
274  41365330             NEW EVERYBODY'S KITCHEN  Staten Island      581   

                   STREET            Street_Address      Zip       Phone  \
40          TARGEE STREET         480 TARGEE STREET  10304.0  3475266458   
156        WATCHOGUE ROAD        280 WATCHOGUE ROAD  10314.0  7189836715   
162  PORT RICHMOND AVENUE  463 PORT RICHMOND AVENUE  10302.0  9178300084   
236            CLOVE ROAD           1150 CLOVE ROAD  10301.0  7184423600   
274        MIDLAND AVENUE        581 MIDLAND AVENUE  10306.0  7189878888   

    Cuisine_Description Inspection_Date  ... Record_Date  \
40              Chicken        1/3/2022  ...   4/30/2022   
156            American       8/21/2017  ...   4/30/2022   
162     Frozen Desserts       2/10/2020  ...   4/30/2022   
236            American       11/6/2019  ...   4/30/2022   
274             Chinese       1/10/2022  ...   4/30/2022   

                                   Inspection_Type   Latitude  Longitude  \
40           Cycle Inspection / Initial Inspection  40.615466 -74.084691   
156               Cycle Inspection / Re-inspection  40.616138 -74.138702   
162  Pre-permit (Operational) / Initial Inspection  40.631149 -74.138813   
236          Cycle Inspection / Initial Inspection  40.618259 -74.105183   
274          Cycle Inspection / Initial Inspection  40.572232 -74.092954   

     Community_District Council_District Census_Tract        BIN  \
40                501.0             49.0       2900.0  5016385.0   
156               501.0             50.0      18901.0  5101265.0   
162               501.0             49.0      21300.0  5024786.0   
236               501.0             49.0      14700.0  5112426.0   
274               502.0             50.0      11202.0  5054740.0   

              BBL   NTA  
40   5.006470e+09  SI37  
156  5.004670e+09  SI07  
162  5.010480e+09  SI28  
236  5.003190e+09  SI07  
274  5.038110e+09  SI45  

[5 rows x 27 columns]

In [89]:
staten_island = pd.DataFrame(si_df)

In [90]:
staten_island

CAMIS                     Restaurant_Name        Borough BUILDING  \
40      50106984               KENNEDY FRIED CHICKEN  Staten Island      480   
156     40387243                JIMMY MAX RESTAURANT  Staten Island      280   
162     50094799  BOTANAS MEXICAN ICE CREAM & SNACKS  Staten Island      463   
236     50016070      THE STONE HOUSE AT CLOVE LAKES  Staten Island     1150   
274     41365330             NEW EVERYBODY'S KITCHEN  Staten Island      581   
...          ...                                 ...            ...      ...   
301639  50084763      FLAMING GRILL & SUPREME BUFFET  Staten Island     1501   
301657  41293635               NO. 1 CHINESE KITCHEN  Staten Island      209   
301666  41646201                        120 BAY CAFE  Staten Island      120   
301686  40542345                  TOWNE DELI & PIZZA  Staten Island     5373   
301689  50057644                ZABROSURA RESTAURANT  Staten Island     2110   

                      STREET            Street_Address      Zip       Phone  \
40             TARGEE STREET         480 TARGEE STREET  10304.0  3475266458   
156           WATCHOGUE ROAD        280 WATCHOGUE ROAD  10314.0  7189836715   
162     PORT RICHMOND AVENUE  463 PORT RICHMOND AVENUE  10302.0  9178300084   
236               CLOVE ROAD           1150 CLOVE ROAD  10301.0  7184423600   
274           MIDLAND AVENUE        581 MIDLAND AVENUE  10306.0  7189878888   
...                      ...                       ...      ...         ...   
301639         FOREST AVENUE        1501 FOREST AVENUE  10302.0  7184476888   
301657  PORT RICHMOND AVENUE  209 PORT RICHMOND AVENUE  10302.0  7188161836   
301666            BAY STREET            120 BAY STREET  10301.0  7182737770   
301686      ARTHUR KILL ROAD     5373 ARTHUR KILL ROAD  10307.0  7182271985   
301689              CLOVE RD             2110 CLOVE RD  10305.0  7184428700   

       Cuisine_Description Inspection_Date  ... Record_Date  \
40                 Chicken        1/3/2022  ...   4/30/2022   
156               American       8/21/2017  ...   4/30/2022   
162        Frozen Desserts       2/10/2020  ...   4/30/2022   
236               American       11/6/2019  ...   4/30/2022   
274                Chinese       1/10/2022  ...   4/30/2022   
...                    ...             ...  ...         ...   
301639             Chinese       3/16/2022  ...   4/30/2022   
301657             Chinese       7/17/2018  ...   4/30/2022   
301666            American       6/17/2019  ...   4/30/2022   
301686               Pizza       2/24/2020  ...   4/30/2022   
301689      Latin American       2/12/2019  ...   4/30/2022   

                                      Inspection_Type   Latitude  Longitude  \
40              Cycle Inspection / Initial Inspection  40.615466 -74.084691   
156                  Cycle Inspection / Re-inspection  40.616138 -74.138702   
162     Pre-permit (Operational) / Initial Inspection  40.631149 -74.138813   
236             Cycle Inspection / Initial Inspection  40.618259 -74.105183   
274             Cycle Inspection / Initial Inspection  40.572232 -74.092954   
...                                               ...        ...        ...   
301639          Cycle Inspection / Initial Inspection  40.624292 -74.138665   
301657               Cycle Inspection / Re-inspection  40.636204 -74.134630   
301666          Cycle Inspection / Initial Inspection  40.639157 -74.075648   
301686          Cycle Inspection / Initial Inspection  40.513369 -74.248012   
301689               Cycle Inspection / Re-inspection  40.603154 -74.083112   

        Community_District Council_District Census_Tract        BIN  \
40                   501.0             49.0       2900.0  5016385.0   
156                  501.0             50.0      18901.0  5101265.0   
162                  501.0             49.0      21300.0  5024786.0   
236                  501.0             49.0      14700.0  5112426.0   
274                  502.0             5

In [91]:
si_df.Cuisine_Description.value_counts()

American                          1851
Pizza                             1218
Chinese                            948
Italian                            944
Mexican                            584
Japanese                           529
Donuts                             281
Sandwiches                         273
Hamburgers                         228
Bakery Products/Desserts           220
Coffee/Tea                         207
Chicken                            188
Frozen Desserts                    180
Asian/Asian Fusion                 146
Spanish                            142
Indian                             133
Latin American                     114
Tex-Mex                            103
Mediterranean                      100
Barbecue                            97
Juice, Smoothies, Fruit Salads      95
Other                               72
Bagels/Pretzels                     72
Greek                               67
Thai                                62
Caribbean                

In [92]:
si_cuisine = pd.DataFrame(si_df.Cuisine_Description.value_counts().reset_index())

In [93]:
si_cuisine

index  Cuisine_Description
0                         American                 1851
1                            Pizza                 1218
2                          Chinese                  948
3                          Italian                  944
4                          Mexican                  584
5                         Japanese                  529
6                           Donuts                  281
7                       Sandwiches                  273
8                       Hamburgers                  228
9         Bakery Products/Desserts                  220
10                      Coffee/Tea                  207
11                         Chicken                  188
12                 Frozen Desserts                  180
13              Asian/Asian Fusion                  146
14                         Spanish                  142
15                          Indian                  133
16                  Latin American                  114
17                         Tex-Mex                  103
18                   Mediterranean                  100
19                        Barbecue                   97
20  Juice, Smoothies, Fruit Salads                   95
21                           Other                   72
22                 Bagels/Pretzels                   72
23                           Greek                   67
24                            Thai                   62
25                       Caribbean                   55
26                        Peruvian                   55
27                         Seafood                   52
28                          German                   51
29       Not Listed/Not Applicable                   50
30                          Polish                   43
31                  Middle Eastern                   36
32                         Russian                   32
33                Eastern European                   31
34         Soups/Salads/Sandwiches                   30
35                        Filipino                   30
36                       Soul Food                   29
37                   Jewish/Kosher                   24
38               Bottled Beverages                   24
39                         Turkish                   23
40                           Irish                   21
41                      Steakhouse                   21
42                         African                   14
43                 Southeast Asian                   14
44                           Cajun                   10
45                      Vegetarian                    9
46                           Tapas                    9
47                Pancakes/Waffles                    9
48  Sandwiches/Salads/Mixed Buffet                    9
49                     Continental                    7
50                Chinese/Japanese                    6
51                Hotdogs/Pretzels                    4

In [94]:
si_cuisine.columns = ['Cuisine Type', 'Restaurants']

In [95]:
si_cuisine

Cuisine Type  Restaurants
0                         American         1851
1                            Pizza         1218
2                          Chinese          948
3                          Italian          944
4                          Mexican          584
5                         Japanese          529
6                           Donuts          281
7                       Sandwiches          273
8                       Hamburgers          228
9         Bakery Products/Desserts          220
10                      Coffee/Tea          207
11                         Chicken          188
12                 Frozen Desserts          180
13              Asian/Asian Fusion          146
14                         Spanish          142
15                          Indian          133
16                  Latin American          114
17                         Tex-Mex          103
18                   Mediterranean          100
19                        Barbecue           97
20  Juice, Smoothies, Fruit Salads           95
21                           Other           72
22                 Bagels/Pretzels           72
23                           Greek           67
24                            Thai           62
25                       Caribbean           55
26                        Peruvian           55
27                         Seafood           52
28                          German           51
29       Not Listed/Not Applicable           50
30                          Polish           43
31                  Middle Eastern           36
32                         Russian           32
33                Eastern European           31
34         Soups/Salads/Sandwiches           30
35                        Filipino           30
36                       Soul Food           29
37                   Jewish/Kosher           24
38               Bottled Beverages           24
39                         Turkish           23
40                           Irish           21
41                      Steakhouse           21
42                         African           14
43                 Southeast Asian           14
44                           Cajun           10
45                      Vegetarian            9
46                           Tapas            9
47                Pancakes/Waffles            9
48  Sandwiches/Salads/Mixed Buffet            9
49                     Continental            7
50                Chinese/Japanese            6
51                Hotdogs/Pretzels            4

American restaurants make up most of the restaurants in this borough.

In [96]:
si_df.Grade.value_counts()

A    4202
B     487
C     168
N     164
Z      63
P      43
Name: Grade, dtype: int64

In [97]:
si_grades = pd.DataFrame(si_df.Grade.value_counts().reset_index())

In [98]:
si_grades

index  Grade
0     A   4202
1     B    487
2     C    168
3     N    164
4     Z     63
5     P     43

In [99]:
si_grades.columns = ['Grade', 'Total']

In [100]:
si_grades

Grade  Total
0     A   4202
1     B    487
2     C    168
3     N    164
4     Z     63
5     P     43

The majority of restaurants in this borough recieved a grade of A from the health inspector.

Now let's do the community districts.

In [ ]:
top5_healthycd = restaurants_df[restaurants_df['Community_District'].isin([111.0, 205.0, 206.0,305.0, 316.0])]

In [ ]:
top5_healthycd.head()

In [ ]:
Top = pd.DataFrame(top5_healthycd)

In [ ]:
Top

In [ ]:
top5_healthycd.Community_District.value_counts()

In [ ]:
top_cds = pd.DataFrame(top5_healthycd.Community_District.value_counts().reset_index())

In [ ]:
top_cds

In [ ]:
top_cds.columns = ['Community District', 'Restaurants']

In [ ]:
top_cds

Manhattan Community District 11 has the most restaurants out of the top 5 healthy community districts.

In [ ]:
top5_healthycd.Cuisine_Description.value_counts()

In [ ]:
top_cuisine = pd.DataFrame(top5_healthycd.Cuisine_Description.value_counts().reset_index())

In [ ]:
top_cuisine

In [ ]:
top_cuisine.columns = ['Cuisine Type', 'Restaurants']

In [ ]:
top_cuisine

Chinese restaurants make up most of the restaurants in these community districts.

In [ ]:
top5_healthycd.Grade.value_counts()

In [ ]:
top_grades = pd.DataFrame(top5_healthycd.Grade.value_counts().reset_index())

In [ ]:
top_grades

In [ ]:
top_grades.columns = ['Grade', 'Total']

In [ ]:
top_grades

The majority of restaurants in this community district recieved a grade of A from the health inspector.

In [ ]:
top5_healthycd.Violation_Description.value_counts()

In [ ]:
top_violations = pd.DataFrame(top5_healthycd.Violation_Description.value_counts().reset_index())

In [ ]:
top_violations 

In [ ]:
top_violations.columns = ['Violation', 'Total']

In [ ]:
top_violations

Non-food contact surface improperly constructed is the most common violation

In [ ]:
top5_healthycd.Violation_Code.value_counts()

In [ ]:
top_vcode = pd.DataFrame(top5_healthycd.Violation_Code.value_counts().reset_index())

In [ ]:
top_vcode 

In [ ]:
top_vcode.columns = ['Violation Code', 'Total']

In [ ]:
top_vcode

Violation Code 10F is the most common violation

In [ ]:
top5_healthycd.Critical_Flag.value_counts()

In [ ]:
top_critical = pd.DataFrame(top5_healthycd.Critical_Flag.value_counts().reset_index())

In [ ]:
top_critical

In [ ]:
top_critical.columns = ['Critical Flag', 'Total']

In [ ]:
top_critical

Most of the violations were critical

In [ ]:
No_facilitiescd = restaurants_df[restaurants_df['Community_District'].isin([318.0, 410.0, 411.0, 503.0])]

In [ ]:
No_facilitiescd.head()

In [ ]:
Bottom = pd.DataFrame(No_facilitiescd)

In [ ]:
Bottom

In [ ]:
No_facilitiescd.Community_District.value_counts()

In [ ]:
bottom_cds = pd.DataFrame(No_facilitiescd.Community_District.value_counts().reset_index())

In [ ]:
bottom_cds

In [ ]:
bottom_cds.columns = ['Community District', 'Restaurants']

In [ ]:
bottom_cds

Queens Community District 11 has the most restaurants out of the bottom (no facility) community districts.

In [ ]:
No_facilitiescd.Cuisine_Description.value_counts()

In [ ]:
bottom_cuisine = pd.DataFrame(No_facilitiescd.Cuisine_Description.value_counts().reset_index())

In [ ]:
bottom_cuisine

In [ ]:
bottom_cuisine.columns = ['Cuisine Type', 'Restaurants']

In [ ]:
bottom_cuisine

American restaurants make up most of the restaurants in these community districts.

In [ ]:
No_facilitiescd.Grade.value_counts()

In [ ]:
bottom_grades = pd.DataFrame(No_facilitiescd.Grade.value_counts().reset_index())

In [ ]:
bottom_grades

In [ ]:
bottom_grades.columns = ['Grade', 'Total']

In [ ]:
bottom_grades

The majority of restaurants in this community district recieved a grade of A  from the health inspector.

In [ ]:
No_facilitiescd.Violation_Description.value_counts()

In [ ]:
bottom_violation = pd.DataFrame(No_facilitiescd.Violation_Description.value_counts().reset_index())

In [ ]:
bottom_violation

In [ ]:
bottom_violation.columns = ['Violation', 'Total']

In [ ]:
bottom_violation

Non-food contact surface improperly constructed is the most common violation

In [ ]:
No_facilitiescd.Violation_Code.value_counts()

In [ ]:
bottom_vcode = pd.DataFrame(No_facilitiescd.Violation_Code.value_counts().reset_index())

In [ ]:
bottom_vcode

In [ ]:
bottom_vcode.columns = ['Violation Code', 'Total']

In [ ]:
bottom_vcode

Violation Code 10F is the most common violation

In [ ]:
No_facilitiescd.Critical_Flag.value_counts()

In [ ]:
bottom_critical = pd.DataFrame(No_facilitiescd.Critical_Flag.value_counts().reset_index())

In [ ]:
bottom_critical

In [ ]:
bottom_critical.columns = ['Critical Flag', 'Total']

In [ ]:
bottom_critical

Most of the violations were critical

Let's make excel workbooks out of the dataframes we created

In [ ]:
xlwriter = pd.ExcelWriter('Restaurants_Output.xlsx')  
boro_restaurants.to_excel(xlwriter, sheet_name='Total_Restaurants_Borough')
CD_restaurants.to_excel(xlwriter, sheet_name='Total_Restaurants_CD')
cuisine_nyc.to_excel(xlwriter, sheet_name='Cuisine_Type_NYC')
restaurant_grade_nyc.to_excel(xlwriter, sheet_name='Restaurant_Grades_NYC')
brooklyn_cuisine.to_excel(xlwriter, sheet_name='Brooklyn_Cuisine')
brooklyn_grades.to_excel(xlwriter, sheet_name='Brooklyn_Grades')
bronx_cuisine.to_excel(xlwriter, sheet_name='Bronx_Cuisine')
bronx_grades.to_excel(xlwriter, sheet_name='Bronx_Grades')
manhattan_cuisine.to_excel(xlwriter, sheet_name='Manhattan_Cuisine')
manhattan_grades.to_excel(xlwriter, sheet_name='Manhattan_Grades')
queens_cuisine.to_excel(xlwriter, sheet_name='Queens_Cuisine')
queens_grades.to_excel(xlwriter, sheet_name='Queens_Grades')
si_cuisine.to_excel(xlwriter, sheet_name='SI_Cuisine')
si_grades.to_excel(xlwriter, sheet_name='Queens_Grades')
xlwriter.close()

In [ ]:
borowriter = pd.ExcelWriter('Restaurants_Output_Boro.xlsx')  
brooklyn.to_excel(borowriter, sheet_name='Brooklyn')
bronx.to_excel(borowriter, sheet_name='Bronx')
manhattan.to_excel(borowriter, sheet_name='Manhattan')
queens.to_excel(borowriter, sheet_name='Queens')
staten_island.to_excel(borowriter, sheet_name='Staten_Island')
borowriter.close()

In [ ]:
cdswriter = pd.ExcelWriter('Restaurants_Output_CDs.xlsx') 
top_cuisine.to_excel(cdswriter, sheet_name='top_cuisine')
top_grades.to_excel(cdswriter, sheet_name='top_grades')
bottom_cuisine.to_excel(cdswriter, sheet_name='bottom_cuisine')
bottom_grades.to_excel(cdswriter, sheet_name='bottom_grades')
top_violations.to_excel(cdswriter, sheet_name='top_violations')
top_critical.to_excel(cdswriter, sheet_name='top_critical')
bottom_violation.to_excel(cdswriter, sheet_name='bottom_violation')
bottom_critical.to_excel(cdswriter, sheet_name='bottom_critical')
top_cds.to_excel(cdswriter, sheet_name='top_cd_restaurants')
bottom_cds.to_excel(cdswriter, sheet_name='bottom_cd_restaurants')
top_vcode.to_excel(cdswriter, sheet_name='top_violation_codes')
bottom_vcode.to_excel(cdswriter, sheet_name='bottom_violation_codes')
cdswriter.close()

In [ ]:
tbcdwriter = pd.ExcelWriter('Restaurants_Output_TBCD.xlsx') 
Top.to_excel(tbcdwriter, sheet_name='Top_CDs')
Bottom.to_excel(tbcdwriter, sheet_name='Bottom_CDs')
tbcdwriter.close()